### PR TITLE
samples: nrf_desktop: Optimize RAM usage

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-276

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

**Comment by Qbicz** [comment]
Did you get the HIDS to print the usage or are the values empirical?

**Comment by pdunaj** [comment]
@Qbicz now they are from stats ;)

File: samples/nrf_desktop/configuration/zephyr.conf [hunk 1]
```diff
@@ -1,9 +1,18 @@
 ################################################################################
 # Zephyr Configuration
 
-CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2200
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=1024
+CONFIG_ISR_STACK_SIZE=760
```
Comment (pdunaj) [inline samples/nrf_desktop/configuration/zephyr.conf]: `<err> bt_ctlr_llsw_ctrl: assert: '_radio.ticker_id_prepare == (7 + conn->handle)' failed`
@wiminordic observed it on CI. I have seen it on Friday but not today (at least not on normal operation).

I am positive this is caused by the overflow on ISR stack as I marked the isr_stack area and observed the usage.
It is very easy to replicated when you stop the CPU with gdb and keep it for some time. When execution continues FW crashes and stack area looks like Verdun after the battle.
The default stack is 2200 but even if I increase it to 4k the stack is still overflown (if I hold the CPU with gdb). It looks like ble stack has some recursion (in mayfly?) that can consume stack of any size if there is a delay of execution of interrupts.

On normal operation stack usage seems to be below 1k.

@cvinayak Do you know if this recursion can be eliminated? We need to reduce the stack size to save RAM.

**Comment by pdunaj** [comment]
@wiminordic I have increated the ISR stack. Could you check if you still see the problems?

**Comment by wiminordic** [comment]
> @wiminordic I have increated the ISR stack. Could you check if you still see the problems?

No, CI did could not reproduce it this time.

**Comment by pdunaj** [comment]
@pizi-nordic , could you merge?

**Comment by pizi-nordic** [comment]
please resolve conflicts

**Comment by pdunaj** [comment]
rebased, @Qbicz could you have a quick look?

**Pull request merged**